lib/email-client.js

```diff
@@ -70,4 +70,19 @@ export default class EmailClient {
 
 		return d.promise;
 	}
+
+	getNewEmailsByRecipient( emailAddress ) {
```

Comment:
Super minor, feel free to ignore how about something like that:
```js
getNewEmailsByRecipient( emailAddress ) {
	var self = this;
	return this.pollEmailsByRecipient( emailAddress )
		.then( function( emails ) {
			return self.deleteAllEmail().then( function () { 
				return emails;
			} );
	         } );
} );
```

or we don't care about delete errors:

```js
getNewEmailsByRecipient( emailAddress ) {
	var self = this;
	return this.pollEmailsByRecipient( emailAddress )
		.then( function( emails ) {
		      self.deleteAllEmail();
                      return emails;
	         } );
} );
```

also, why can't we use arrow functions here? we seems to use the ES6 export / import syntax here?